src/codeu/chat/client/maingui/LoginPanel.java

```diff
@@ -0,0 +1,72 @@
+package codeu.chat.client.maingui;
+import java.awt.*;
+import java.awt.event.ActionListener;
+import java.awt.event.ActionEvent;
+import javax.swing.*;
+
+import codeu.chat.client.ClientContext;
+import codeu.chat.common.LoginInputCallback;
+
+@SuppressWarnings("serial")
+public final class LoginPanel extends JPanel{
+
+    private final ClientContext clientContext;
+    private final LoginInputCallback loginInputCallback;
+
+    public LoginPanel(ClientContext clientContext, LoginInputCallback loginInputCallback) {
+        super();
+        this.clientContext = clientContext;
+        this.loginInputCallback = loginInputCallback;
+        initialize();
+    }
+    private void initialize() {
+
+        // This panel contains the login form for get access to the chat.
+
+        // Configure the size of the panel and the main structure.
+        this.setLayout(new GridBagLayout());
+        this.setPreferredSize(new Dimension(400,500));
+        GridBagConstraints panelC = new GridBagConstraints();
```

Comment:
Why is it called "panelC"?  What are panels A and B?

Oh, it's probably meant for "panel constraints"?  I'd call it "mainPanel" or "panelConstraints" or something.